pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -734,6 +733,30 @@ public int getConcurrency() throws SQLException {
     return getTimestampUtils().toLocalDateTime(string);
   }
 
+  private @Nullable LocalDate getLocalDate(int i) throws SQLException {
+    byte[] value = getRawValue(i);
+    if (value == null) {
+      return null;
+    }
+
+    if (isBinary(i)) {
+      int col = i - 1;
+      int oid = fields[col].getOID();
+      if (oid == Oid.DATE) {
+        return getTimestampUtils().toLocalDateBin(value);
+      } else if (oid == Oid.TIMESTAMP) {
+        return getTimestampUtils().toLocalDateTimeBin(value).toLocalDate();
+      } else {
```

Comment:
So why would we not want to return a localdate from a timestamp with time zone ?

Reply:
The original code line 3699 did not supported this, too.

But I fully aggree, we can add it, is a trivial addition similar to this one. If we support it here, we must add it in line 372, too.

Reply:
In `getLocalDateTime()` above it also does not support it.

Reply:
Well this is a question for debate. The fact that it wasn't supported is somewhat orthogonal

Reply:
At other places it says "xy is required by JDBC spec", maybe it did just implement the minimal set.

In my opinion the main getObject() method can easily support all types with and without offset. The problem with the TZ ones is that it is unclear to me if it should just strip time zone information or use default timezone during the conversion. But I haven't read the JDBC spec how this should work. So better throw exception instead of returning wrong values.

Reply:
As said in my previous post the whole `getObject()` is also missing `OffsetTime` from `TIMETZ`. This would be a really nice addition.